src/Renderer/ThreeExtended/Quaternion.js

```diff
@@ -0,0 +1,56 @@
+import * as THREE from 'three';
+
+if (!THREE.Quaternion.prototype.preciseSlerp) {
+    THREE.Quaternion.prototype.preciseSlerp = function _preciseSlerp(qb, t) {
```

Comment:
This function looks very similar to [the one from three](https://github.com/mrdoob/three.js/blob/dev/src/math/Quaternion.js#L499).

Could you explain what is the difference?

Reply:
https://github.com/mrdoob/three.js/blob/dev/src/math/Quaternion.js#L538 : 

If the angle is small, it averages the values of the quaternion

Reply:
Ok.
Then please add some comments to your implementation stating that it's a copy-paste from three with this modification.
And a justification of why this modification is needed - because from an outside perspective I don't see any harm in averaging the quaternion.

Especially since the original algorithm explicitely says:
`When theta is 180 degrees the result is undefined because there is no shortest direction to rotate`.
So if I'm not wrong your solution isn't more "precise", it's simply a different implementation choice for the case where theta is ~180°.

WDYT?

Reply:
yes indeed, I commented and I took the condition when theta = 180 °

Reply:
Better thanks.

So, threejs rounds when the angle is `180° ± 0.114°`. And your solution theshold is `180° ± 0.00000114°`.
`0.114°` seems reasonnable, so can you explain why did you need `0.00000114°`?

Also, did you consider opening a PR to fix this issue directly in threejs (either by changing the constant or making it a parameter)?

Reply:
Because the equivalent of 0.001 ° on the surface of the globe is greater than 110 meters, so it's not precise enough

Reply:
But the issue only happen when `theta` (ie: the rotation needed to transform quaternion1 in quaternion2) is almost 180° - that would mean rotating completely around the globe, right? In which case the camera will travel ~20km, so the 110m error is less impressive.

Do you have a reproducable example so I can understand the problem you're solving?

Reply:
You can see this [example](https://github.com/iTowns/itowns/pull/774/files#diff-0936d27ecea5501d163f8546d2615538)

Reply:
Yeah. That's not what I meant :)

I meant: a real example in iTowns where the `slerp` "precision" issue is visible or has a visible effect.

Reply:
By example, slerp is used in globeControls damping, the imprecisions cause error in this feature.

Reply:
I know it's in GlobeControls, since it's the only user of this function. I really need a step-by-step reproducer to understand the necessity of this PR.

You also forgot to answer a couple questions:
- did you consider opening a PR to fix this issue directly in threejs (either by changing the constant or making it a parameter)? (from https://github.com/iTowns/itowns/pull/774#discussion_r192720807)
- the issue only happen when theta (ie: the rotation needed to transform quaternion1 in quaternion2) is almost 180° - that would mean rotating completely around the globe, right?

Reply:
- replace preciseSlerp by slerp in globeControls and try move globe with only left button (being close to the ground). The damping doesn't work properly

- I have not considered for the moment to open a THREE PR 

- that would mean rotating completely around the globe, right?  it's possible, but the unit test also show that small rotations don't give the good result

Reply:
> replace preciseSlerp by slerp in globeControls and try move globe with only left button (being close to the ground). The damping doesn't work properly

AFAICT it's a logic issue in the GlobeControls code. It's doing lerp between 2 nearly identical quaternions.
e.g:
```js
        root.qDelta.preciseSlerp(damp, root.dampingFactor * 0.2);
```
You shouldn't accumulate the change in root.qDelta, but instead make the `t` parameter for `preciseSlerp` vary from 0 to 1 (otherwise you'll end up lerping identical quaternions)

> the unit test also show that small rotations don't give the good result

The first unittest shows that the result is different when testing nearly identical quaternion. But it misses the most useful part: comparing to a known good value (`assert.equal(R.w, known_value_computed_using_another_method`).

Anyway, IMHO this PR isn't necessary: the calling site (GlobeControls) should be fixed instead.

Reply:
I have new PR where I use `slerp` with `t` vary from 0 to 1, and there's the same problem. 
I push a unit test with another method, which shows that the projected error on the globe can reach 500 m

Reply:
> I have new PR where I use slerp with t vary from 0 to 1, and there's the same problem.

Did you push this branch somewhere so I can test it?

>  I push a unit test with another method, which shows that the projected error on the globe can reach 500 m

Did you pushed it?
Because it's the same test, lerping between 2 ~identical quaternions...

Reply:
yes, I [pushed](https://github.com/iTowns/itowns/pull/774/commits/a62d8214c59dc48d02f5124e8da353dc55887c59#diff-0936d27ecea5501d163f8546d2615538R9).
As announced from the beginning this PR, this test highlights the imprecision of the original method, which is also highlighted in `globeControl`. I don't think that  vary `t` from 0 to 1 fix the problem

Reply:
Actually the original paper says the result is undefined if theta is 180° but threejs code tests for sin(theta) ~= 0 which happens when theta is 0° or 180°.

So IMHO: the bug should be fixed upstream (for instance by replacing the if test with `cosHalfTheta <= 0 && Math.abs(sinHalfTheta) < 0.001`).